rpc/client.go

```diff
@@ -109,6 +148,106 @@ func (opts *UploadReportOptions) artifactConsumer(ctx context.Context, testChan
 	}
 }
 
+// getSignedUrl calls the Data Pipes API to retrieve a signed URL, where it can PUT the report json.
+// Data Pipes docs: https://github.com/10gen/data-pipes
+func getSignedURL(task string, execution int, data []byte, AWSAccessKey string, AWSSecretKey string, AWSToken string, dataPipesHost string, dataPipesRegion string) (string, error) {
```

Comment:
Instead of passing in all of these arguments, can you just pass in the `UploadReportOptions` struct?